cmd/k8s-keystone-auth/main.go

```diff
@@ -15,47 +15,45 @@ limitations under the License.
 package main
 
 import (
-	"fmt"
 	"os"
 
+	"github.com/spf13/cobra"
 	"github.com/spf13/pflag"
+	"k8s.io/component-base/cli"
 	"k8s.io/klog/v2"
 
 	"k8s.io/cloud-provider-openstack/pkg/identity/keystone"
 	"k8s.io/cloud-provider-openstack/pkg/version"
-	kflag "k8s.io/component-base/cli/flag"
-	"k8s.io/component-base/logs"
 )
 
+var config = keystone.NewConfig()
+
 func main() {
-	var showVersion bool
-	pflag.BoolVar(&showVersion, "version", false, "Show current version and exit")
+	cmd := &cobra.Command{
+		Use:   "k8s-keystone-auth",
+		Short: "Keystone authentication webhook plugin for Kubernetes",
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := config.ValidateFlags(); err != nil {
+				klog.Errorf("%v", err)
+				os.Exit(1)
+			}
+
+			keystoneAuth, err := keystone.NewKeystoneAuth(config)
+			if err != nil {
+				klog.Errorf("%v", err)
+				os.Exit(1)
+			}
+			keystoneAuth.Run()
+
+		},
+		Version: version.Version,
+	}
 
-	logs.AddFlags(pflag.CommandLine)
 	keystone.AddExtraFlags(pflag.CommandLine)
 
-	logs.InitLogs()
-	defer logs.FlushLogs()
```

Comment:
I wonder whether removing this may result in missing logs or error messages?

Reply:
This should not be necessary as `cli.Run` takes care of this for us [1][2][3]

[1] https://github.com/kubernetes/component-base/blob/v0.28.1/cli/run.go#L45
[2] https://github.com/kubernetes/component-base/blob/v0.28.1/cli/run.go#L88
[3] https://github.com/kubernetes/component-base/blob/v0.28.1/cli/run.go#L90